### PR TITLE
Add check for possible slowloris

### PR DESCRIPTION
**Pull request by cPanelMikeB**
Added a check for the number of established Apache connections as a high number can indicate a slowloris attack.

### CODE REVIEW

File: ssp [hunk 1]
```diff
@@ -8826,6 +8828,22 @@ sub check_for_cpro {
 
 ## END malware checks
 
+sub detect_slowloris {
+    my $sockcount = 75;
```
Comment (davelcpanelnet-zz) [inline ssp:8832]: Why 75?

File: ssp [hunk 2]
```diff
@@ -8826,6 +8828,22 @@ sub check_for_cpro {
...
+        if ( $line =~ m/ESTABLISHED.*httpd/ ) {
+            $count++;
+        }
+    }
+    if ( $count >= $sockcount ) {
+        print_warn("Possible SLOWLORIS attack: Found ".$count." Apache sockets in ESTABLISHED\n");
+        print_warning("\t \\_ See https://documentation.cpanel.net/display/EA/How+To+Mitigate+Slowloris+Attacks");
+    }
+}
+
```
Comment (davelcpanelnet-zz) [inline ssp:8846]: We really would like this to be a more robust check.  We need to assure that the IPS are also unique as SlowLorris tends to always prefer a single IP, correct?

I'm not sure how automatable we can make a slowlorris detection and not constantly throw noise.

File: ssp (hunk 1 above)
Comment (cPanelMikeB) [reply ssp:8832]: Arbitrary based on typical connection limit of 150.

File: ssp (hunk 2 above)
Comment (cPanelMikeB) [reply ssp:8846]: Typically, yes, they are a single IP address. I have, however, seen it come from multiple IP addresses. There are a few IRC botnets that can do it.

File: ssp [hunk 3]
```diff
@@ -550,6 +551,7 @@ sub non_cpanel_checks_only {
     check_for_license_info();
 
     ## [WARN]
+    detect_slowloris();
```
Comment (cPanelChrisD) [inline ssp:554]: This does not need to be run in the non-cPanel checks.

**Comment by davelcpanelnet-zz** [comment]
I need you to resubmit and change the md5sum as well.  

https://github.com/CpanelInc/SSP/blob/master/SHA512SUM

File: ssp [hunk 4]
```diff
@@ -8826,6 +8828,22 @@ sub check_for_cpro {
 
 ## END malware checks
 
+sub detect_slowloris {
+    my $sockcount = 75;
+    my $count = 0;
+
+    return unless my $netstat_out = timed_run( 0, 'netstat', '-nap' );
```
Comment (cPanelChrisD) [inline ssp:8835]: Please use the existing @LSOF_80 which contains the contents of "lsof -n -i 'tcp:80'" instead of running netstat.

File: ssp [hunk 5]
```diff
@@ -8826,6 +8828,22 @@ sub check_for_cpro {
...
+    for my $line ( split( '\n', $netstat_out ) ) {
+        if ( $line =~ m/ESTABLISHED.*httpd/ ) {
+            $count++;
+        }
+    }
+    if ( $count >= $sockcount ) {
+        print_warn("Possible SLOWLORIS attack: Found ".$count." Apache sockets in ESTABLISHED\n");
```
Comment (cPanelChrisD) [inline ssp:8842]: See other checks for how print_warn is generally used only to prefix the output, followed by print_warning to contain the bulk of the output.

File: ssp (hunk 1 above)
Comment (cPanelChrisD) [inline ssp:8832]: Is just checking for >75 established connections the most robust way to check for Slowloris?  Busy servers can hit this mark easily while not under attack.

File: ssp [hunk 6]
```diff
@@ -9,7 +9,7 @@ package SSP;
...
+THE SOFTWARE LICENSED HEREUNDER IS PROVIDED "AS IS" AND CPANEL HEREBY DISCLAIMS ALL WARRANTIES OF ANY KIND, WHETHER EXPRESS OR IMPLIED, RELATING TO THE SOFTWARE, ITS THIRD PARTY COMPONENTS, AND ANY DATA ACCESSED THEREFROM, OR THE ACCURACY, TIMELINESS, COMPLETENESS, OR ADEQUACY OF THE SOFTWARE, ITS THIRD PARTY COMPONENTS, AND ANY DATA ACCESSED THEREFROM, INCLUDING THE IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY, SATISFACTORY QUALITY, FITNESS FOR A PARTICULAR PURPOSE, AND NON-INFRINGEMENT.  CPANEL DOES NOT WARRANT THAT THE SOFTWARE OR ITS THIRD PARTY COMPONENTS ARE ERROR-FREE OR WILL OPERATE WITHOUT INTERRUPTION.  IF THE SOFTWARE, ITS THIRD PARTY COMPONENTS, OR ANY DATA ACCESSED THEREFROM IS DEFECTIVE, YOU ASSUME THE SOLE RESPONSIBILITY FOR THE ENTIRE COST OF ALL REPAIR OR INJURY OF ANY KIND, EVEN IF CPANEL HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DEFECTS OR DAMAGES.  NO ORAL OR WRITTEN INFORMATION OR ADVICE GIVEN BY CPANEL, ITS AFFILIATES, LICENSEES, DEALERS, SUB-LICENSORS, AGENTS OR EMPLOYEES SHALL CREATE A WARRANTY OR IN ANY WAY.
```
Comment (cPanelChrisD) [inline ssp:12]: This is also an unnecessary change.
